xknx/devices/fan.py

```diff
@@ -58,11 +60,28 @@ def __init__(
         self.mode = FanSpeedMode.STEP if max_step else FanSpeedMode.PERCENT
         self.max_step = max_step
 
+        # If there is no dedicated speed GA, then the main GA of the fan serves
+        # as speed GA (and implicitely on/off as well). If however there is a dedicated
+        # speed GA, then the regular state GA becomes an on/off switch.
```

Comment:
Tbh I don't think this is a good concept. The `group_address` attribute would now take GAs with different DPTs depending on wether or not the `group_address_speed` is set. ([Zen of Python](https://peps.python.org/pep-0020/))
It would be easier to understand if they would always expect the same DPT.

In HA I see 2 options then
- we can make it a breaking change and use `address` (optional) for switch and add a new required key `speed_address`
- we can add a new optional key `switch_address` and call it a day

Reply:
In the end I would be fine with both options.
1. If we use that one, I'd say the new key `speed_address` should be optional, since I assume there can be fans that can only be switched on and off, but have not speed control.
2. Fine with me as well. I just proposed the current PR with the intention of keeping a consistent feel across the platforms (with `address` controlling the main on/off state), but I do see the DPT issue with this dynamic approach.

I will adjust the coding to go with option 2, since then we have neither a breaking change nor a dynamic DPT.

Reply:
But they might still be able to oscillate which the fan platform would already support. But since we are not going with option 2, that point from option is moot by now 😄.

Reply:
Oh dear, I seem to have no clue about fans. Please ignore my comments 🤣